src/js/actions/transform.js

```diff
@@ -1119,9 +1120,9 @@ define(function (require, exports) {
                     nextDoc = appStore.getCurrentDocument();
                     this.flux.actions.layers.resetBounds(nextDoc, nextDoc.layers.allSelected);
                 });
-        }.bind(this);
+        }.bind(this), this);
 
-        _moveListener = function () {
+        _moveListener = synchronization.debounce(function () {
```

Comment:
Fair thought, and probably right.

BUT

Given the user is clicking and dragging like crazy, I'd argue that the correctness of the history is not the primary concern of the user.